tests/text/generate/test_integer.py

```diff
@@ -68,3 +69,15 @@ def test_integer_proposal():
             (4, 1),
         ),
     )
+
+    logits = torch.ones(len(model.tokenizer.vocabulary))
+    result = generator.create_proposal(torch.tensor([[4, 0], [2, 2]]), logits)
+    assert torch.equal(
+        result,
+        torch.tensor(
+            [
+                [1.0, -math.inf, -math.inf, -math.inf, -math.inf, -math.inf],
+                [1.0, 1.0, 1.0, -math.inf, 1.0, -math.inf],
+            ]
+        ),
+    )
```

Comment:
This test demonstrates the EOS and padding functionality by only allowing EOS to be sampled for the sequence that has already terminated with EOS (i.e. the first sequence).  It's set to use EOS whenever `model.tokenizer.pad_token` isn't available.  I'm not sure if that's the best approach, but it should be easy to change regardless.

Reply:
We need to generate `eos_token_id` in all cases here. When found the `is_finished` method will mark the current sequence as finished, and the `__call__` method will then stop sampling new tokens for this sequence and insert the `pad_token_id`s.